smaily-for-wp.php

```diff
@@ -89,198 +90,38 @@ function smaily_admin_render() {
-	$config     = $wpdb->get_row( "SELECT * FROM `$table_name`" );
-
-	// Make a opt-in request to server.
-	$server = 'https://' . $config->domain . '.sendsmaily.net/api/opt-in/';
-	$lang   = explode( '-', $params['lang'] );
-	$array  = array(
-		'remote'        => 1,
-		'success_url'   => $redirect_url,
-		'failure_url'   => $redirect_url,
-		'language'      => $lang[0],
-	);
-
-	// Add autoresponder if selected.
-	if ( (int) $config->autoresponder !== 0 ) {
-		$array['autoresponder'] = $config->autoresponder;
-	}
-
-	$form_values = [];
-	// Add custom form values to Api request if available.
-	foreach ( $params as $key => $value ) {
-		if ( array_key_exists( $key, $array ) ) {
-			continue;
-		} else {
-			$form_values[ $key ] = $value;
-		}
-	}
-
-	$array = array_merge( $array, $form_values );
-	require_once( SMLY4WP_PLUGIN_PATH . '/code/Request.php' );
-	$result  = (new Smaily_Plugin_Request())
-		->setUrl($server)
-		->setData($array)
-		->post();
-
-	if ( empty( $result ) ) {
-		$redirect_url = add_query_arg( 'smaily_form_error',
-			rawurlencode( __( 'Something went wrong', 'wp_smaily' ) ), $redirect_url );
-	} elseif ( (int) $result['code'] !== 101 ) {
-		switch ( (int) $result['code'] ) {
-			case 201:
-				$redirect_url = add_query_arg( 'smaily_form_error',
-					rawurlencode( __( 'Form was not submitted using POST method.', 'wp_smaily' ) ), $redirect_url );
-				break;
-			case 204:
-				$redirect_url = add_query_arg( 'smaily_form_error',
-					rawurlencode( __( 'Input does not contain a recognizable email address.', 'wp_smaily' ) ), $redirect_url );
-				break;
-			case 205:
-				$redirect_url = add_query_arg( 'smaily_form_error',
-					rawurlencode( __( 'Could not add to subscriber list for an unknown reason. Probably something in Smaily.', 'wp_smaily' ) ), $redirect_url );
-				break;
-			default:
-				$redirect_url = add_query_arg( 'smaily_form_error',
-					rawurlencode( __( 'Something went wrong', 'wp_smaily' ) ), $redirect_url );
-				break;
-		}
+/**
+ * Render notification bar.
+ *
+ * @return void
+ */
+function smaily_handle_response() {
+	$isset_response = isset( $_GET['code'] ) && isset( $_GET['message'] );
+	$code = $isset_response ? $_GET['code'] : '';
+	switch ( (int) $code ) {
+		case 101:
+			echo '<div id="notifybar">
+					<p>' . esc_html__( 'You have been successfully subscribed.', 'wp_smaily' ) . '</p>
+				 </div>';
+			break;
+
+		case 201:
+			echo '<div id="notifybar">
+					<p>' . esc_html__( 'Data must be posted with POST method.', 'wp_smaily' ) . '</p>
+				 </div>';
+			break;
+
+		case 204:
+			echo '<div id="notifybar">
+					<p>' . esc_html__( 'Input does not contain a recognizable email address.', 'wp_smaily' ) . '</p>
+				 </div>';
+			break;
+
+		case 205:
+			echo '<div id="notifybar">
+					<p>' . esc_html__( 'Could not add to subscriber list for an unknown reason. Probably something in Smaily.', 'wp_smaily' ) . '</p>
+				 </div>';
+			break;
```

Comment:
You can simplify this code:

```php
if ( ! isset( $_GET['code'] ) || ! isset( $_GET['message'] ) ) {
    return;
}

$errorMessage = NULL;
switch ( (int) $_GET['code'] ) {
    case 101:
        $errorMessage = esc_html__( 'You have been successfully subscribed.', 'wp_smaily' );
        break;
    case 201:
        $errorMessage = esc_html__( 'Data must be posted with POST method.', 'wp_smaily' );
        break;
    case 204:
        $errorMessage = esc_html__( 'Input does not contain a recognizable email address.', 'wp_smaily' );
        break;
    default:
        $errorMessage = esc_html__( 'Could not add to subscriber list for an unknown reason. Probably something in Smaily.', 'wp_smaily' );
        break;
}

echo '<div id="notifybar"><p>' . $errorMessage . '</p></div>';
```

With this approach you will save yourself from managing same HTML, and it will bring out what is important in each section (if, switch and echo) of the code.

PS. I also changed the `case 205` to `default` - this will always default to an error message if some unhandled `code` value is returned.

Reply:
Did you also consider using [get_query_var](https://developer.wordpress.org/reference/functions/get_query_var/) function for fetching query arguments?

Reply:
I didn't consider it since fetching query arguments seemed so trivial here. I'll add it since it's WP functionality

Reply:
I can add it but I can't find the pros of it rather than just accessing the values directly, especially since I'm already validating `message` and `code`

Reply:
I think it's mainly used for public query variables: p, paged, etc.
I would keep the current solution.

Reply:
OK